### PR TITLE
Improve build makefile

### PR DESCRIPTION
**Pull request by abhijit-mahajani-imgtec**
Remove building project0 specific led-actuator

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

### CODE REVIEW

**Comment by abhijit-mahajani-imgtec** [comment]
test with jenkins

**Comment by abhijit-mahajani-imgtec** [comment]
build all

**Comment by abhijit-mahajani-imgtec** [comment]
test with jenkins

**Comment by abhijit-mahajani-imgtec** [comment]
test with jenkins

**Comment by nikhil-zinjurde-imgtec** [comment]
Approved

File: Makefile [hunk 1]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
 clean: clean_openwrt clean_feeds clean_contiki clean_binaries
 
 # Building OpenWRT
-openwrt/feeds.conf:
-	cd ../dist/openwrt; \
-	cp ../../build/feeds.conf .; \
+.PHONY: openwrt/feeds.conf
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: This is not required, as no one is calling this anymore

File: Makefile [hunk 2]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
+openwrt/feeds.conf: $(DIR__OPENWRT)/feeds.conf
+
+$(DIR__OPENWRT)/feeds.conf:
+	cd $(DIR__OPENWRT); \
+	cp $(DIR__BUILD)/feeds.conf .; \
 	./scripts/feeds update -a; \
-	./scripts/feeds install -a; \
+	./scripts/feeds install -a;
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: any reason for removing backslash at the end?

File: Makefile [hunk 3]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
 ifneq (_,_$(findstring all,$P))
-	cd feeds/packages; patch -p1 < ../../../../build/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
-	patch -p1 < ../../../../build/0001-node-host-turn-off-verbose.patch;
+	cd $(DIR__OPENWRT)/feeds/packages; patch -p1 < $(DIR__BUILD)/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
+	patch -p1 < $(DIR__BUILD)/0001-node-host-turn-off-verbose.patch;
 endif
 
+.PHONY: openwrt/.config
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: we could get rid of this too

File: Makefile [hunk 4]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
+.PHONY: openwrt/version
+openwrt/version: $(DIR__OPENWRT)/version
 
-openwrt/version:
-	./getver.sh ../dist/openwrt > ../dist/openwrt/version
+$(DIR__OPENWRT)/version:
+	./getver.sh  $(DIR__OPENWRT) > $(DIR__OPENWRT)/version
 
 .PHONY: build_openwrt
 build_openwrt: openwrt/.config openwrt/version
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: why not directly use $(DIR__OPENWRT)/.config ?

File: Makefile [hunk 5]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
 clean_openwrt:
-	$(MAKE) -C ../dist/openwrt clean
+	$(MAKE) -C $(DIR__OPENWRT) clean
 
 # Clean Contiki
 .PHONY: clean_contiki
 clean_contiki:
-	$(MAKE) -C ../packages/led-actuator TARGET=mikro-e clean
-	$(MAKE) -C ../packages/button-sensor TARGET=mikro-e clean
+	$(MAKE) -C $(DIR__CKT)/packages/button-sensor TARGET=mikro-e clean
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: wouldn't it be better to do distclean ?

File: Makefile [hunk 6]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
 clean: clean_openwrt clean_feeds clean_contiki clean_binaries
 
 # Building OpenWRT
-openwrt/feeds.conf:
-	cd ../dist/openwrt; \
-	cp ../../build/feeds.conf .; \
+.PHONY: openwrt/feeds.conf
+openwrt/feeds.conf: $(DIR__OPENWRT)/feeds.conf
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: This is not required, as no one is calling this anymore

File: Makefile [hunk 7]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
+	./scripts/feeds install -a;
 ifneq (_,_$(findstring all,$P))
-	cd feeds/packages; patch -p1 < ../../../../build/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
-	patch -p1 < ../../../../build/0001-node-host-turn-off-verbose.patch;
+	cd $(DIR__OPENWRT)/feeds/packages; patch -p1 < $(DIR__BUILD)/0001-glib2-make-libiconv-dependent-on-ICONV_FULL-variable.patch; \
+	patch -p1 < $(DIR__BUILD)/0001-node-host-turn-off-verbose.patch;
 endif
 
+.PHONY: openwrt/.config
+openwrt/.config: $(DIR__OPENWRT)/.config
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: we could get rid of this too

File: Makefile [hunk 8]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
 
-openwrt/.config: openwrt/feeds.conf
+$(DIR__OPENWRT)/.config: $(DIR__OPENWRT)/feeds.conf
 	if test $(findstring P=,$(MAKEFLAGS)) && test -f $P; then \
-		cat $P > ../dist/openwrt/.config; \
+		cat $P > $(DIR__OPENWRT)/.config; \
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: cat $P > $<

File: Makefile [hunk 9]
```diff
@@ -11,77 +17,83 @@ openwrt: build_openwrt copy_openwrt
...
-	$(MAKE) -C ../dist/openwrt defconfig
+	$(MAKE) -C $(DIR__OPENWRT) defconfig
+
+.PHONY: openwrt/version
+openwrt/version: $(DIR__OPENWRT)/version
 
-openwrt/version:
-	./getver.sh ../dist/openwrt > ../dist/openwrt/version
+$(DIR__OPENWRT)/version:
+	./getver.sh  $(DIR__OPENWRT) > $(DIR__OPENWRT)/version
```
Comment (mayank-sirotiya-imgtec) [inline Makefile]: use $@

**Pull request merged**